source-git-monorepos/README.md

```diff
@@ -1,41 +1,80 @@
-# Support monorepos in source-git
+# Support for Monorepos
 
-Monorepos store the source code of multiple projects in a single
-repository. They are a special use-case from a source-git point of view,
-because an upstream monorepo maps to multiple dist-git repositories in the
-distribution.
+Monorepos store the source code of multiple projects in a single repository.
+They are a special use-case from a source-git point of view, because an
+upstream monorepo maps to multiple dist-git repositories in the distribution.
 
 One example for the above is [LLVM] from which code flows into multiple
 dist-git repositories in Fedora Linux.
 
-Currently the source-git commands in source-git map an upstream repo to a
-single dist-git repo in the distribution.
+Currently Packit maps an upstream repo to a single dist-git repo in the
+distribution.
 
 To change this, a mechanism to specify which path(s) in the monorepo to map to
-which dist-git repo in the distribution should be introduced.
+which dist-git repo in the distribution needs to be introduced.
+
+This could be done by introducing a `packages` key, to hold a dictionary of
+`{package_name: package_object}`. Package-objects whould have keys identical
+to the current top-level keys in `.packit.yaml` and `source-git.yaml`, except
+`upstream_project_url` and `upstream_ref`—these are left to be top-level keys
+only, and a new `package_paths` key which is a list of paths in the monorepo
+which map to the dist-git repo specified by `downstream_package_name`. If
+`downstream_package_name` is not specified, it would be assumed that the
+dist-git repo is called `package_name`.
+
+When processing a package from a monorepo, Packit should operate within these
+paths, that is, limit patch generation from Git-history to these paths.
+
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
```

Comment:
(definitely can't tell the editor :eyes:)

Reply:
:see_no_evil: